File: engine/engine-model/src/main/scala/tools/packet/character/npc/RemovePlayerNPC.scala
package tools.packet.character.npc

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class RemovePlayerNPC(private var _objectId: Int) extends PacketInput {
  def objectId: Int = _objectId

  override def opcode(): SendOpcode = SendOpcode.IMITATED_NPC_DATA
}

File: engine/engine-model/src/main/java/tools/packet/character/npc/RemovePlayerNPC.java
package tools.packet.character.npc;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record RemovePlayerNPC(Integer objectId) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.IMITATED_NPC_DATA;
   }
}